src/dmd/backend/debugprint.d

```diff
@@ -0,0 +1,486 @@
+void WROP(uint oper)
+{
+  if (oper >= OPMAX)
+  {     printf("op = x%x, OPMAX = %d\n",oper,OPMAX);
+        assert(0);
+  }
+  ferr(debtab[oper]);
+  ferr(" ");
+}
+
+/*******************************
+ * Write TYxxxx
+ */
+
+void WRTYxx(tym_t t)
+{
+    if (t & mTYnear)
+        printf("mTYnear|");
+    if (t & mTYfar)
+        printf("mTYfar|");
+    if (t & mTYcs)
+        printf("mTYcs|");
+    if (t & mTYconst)
+        printf("mTYconst|");
+    if (t & mTYvolatile)
+        printf("mTYvolatile|");
+//#if !MARS && (__linux__ || __APPLE__ || __FreeBSD__ || __OpenBSD__ || __sun)
+//    if (t & mTYtransu)
+//        printf("mTYtransu|");
+//#endif
+    t = tybasic(t);
+    if (t >= TYMAX)
+    {   printf("TY %x\n",cast(int)t);
+        assert(0);
+    }
+    printf("TY%s ",tystring[tybasic(t)]);
+}
+
+void WRBC(uint bc)
+{
+    __gshared const char[7][BCMAX] bcs =
+        ["unde  ","goto  ","true  ","ret   ","retexp",
+         "exit  ","asm   ","switch","ifthen","jmptab",
+         "try   ","catch ","jump  ",
+         "_try  ","_filte","_final","_ret  ","_excep",
+         "jcatch","_lpad ",
+        ];
+
+    assert(bc < BCMAX);
+    printf("BC%s",bcs[bc].ptr);
+}
+
+/************************
+ * Write arglst
+ */
+
+void WRarglst(list_t a)
+{ int n = 1;
+
+  if (!a) printf("0 args\n");
+  while (a)
+  {     const(char)* c = cast(const(char)*)list_ptr(a);
+        printf("arg %d: '%s'\n", n, c ? c : "NULL");
+        a = a.next;
+        n++;
+  }
+}
+
+/***************************
+ * Write out equation elem.
+ */
+
+void WReqn(elem *e)
+{ __gshared int nest;
+
+  if (!e)
+        return;
+  if (OTunary(e.Eoper))
+  {
+        WROP(e.Eoper);
+        if (OTbinary(e.EV.E1.Eoper))
+        {       nest++;
+                ferr("(");
+                WReqn(e.EV.E1);
+                ferr(")");
+                nest--;
+        }
+        else
+                WReqn(e.EV.E1);
+  }
+  else if (e.Eoper == OPcomma && !nest)
+  {     WReqn(e.EV.E1);
+        printf(";\n\t");
+        WReqn(e.EV.E2);
+  }
+  else if (OTbinary(e.Eoper))
+  {
+        if (OTbinary(e.EV.E1.Eoper))
+        {       nest++;
+                ferr("(");
+                WReqn(e.EV.E1);
+                ferr(")");
+                nest--;
+        }
+        else
+                WReqn(e.EV.E1);
+        ferr(" ");
+        WROP(e.Eoper);
+        if (e.Eoper == OPstreq)
+            printf("%d", cast(int)type_size(e.ET));
+        ferr(" ");
+        if (OTbinary(e.EV.E2.Eoper))
+        {       nest++;
+                ferr("(");
+                WReqn(e.EV.E2);
+                ferr(")");
+                nest--;
+        }
+        else
+                WReqn(e.EV.E2);
+  }
+  else
+  {
+        switch (e.Eoper)
+        {   case OPconst:
+                elem_print_const(e);
+                break;
+            case OPrelconst:
+                ferr("#");
+                goto case OPvar;
+
+            case OPvar:
+                printf("%s",e.EV.Vsym.Sident.ptr);
```

Comment:
This looks a bit dangerous. It doesn't look `char[1]` is null-terminated in the default cause, e.g. https://run.dlang.io/is/VHJOtq

Maybe "%c" or being length-specific with `%.*s`?

Reply:
Sident is always 0 terminated, that's why the original being just Sident worked.